caas/kubernetes/clientconfig/k8s.go

```diff
@@ -15,15 +16,18 @@ import (
 
 var logger = loggo.GetLogger("juju.caas.kubernetes.clientconfig")
 
```

Comment:
Please include a comment explaining that the reader passed in is used to read the config. The part of the comment that it "parses Kubernetes client configuration from the default location or $KUBECONFIG." is not valid now that we are passing in the reader.